examples/shell-environments/flake.nix

```diff
@@ -0,0 +1,27 @@
+{
+  description = "Description for the project";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  };
+
+  outputs = inputs@{ flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = [ "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ];
+      perSystem = { config, self', inputs', pkgs, system, ... }: {
+        devShells.default = pkgs.mkShell {
+          buildInputs = with pkgs; [ terraform wget bat git ];
+        };
+
+        devShells.another_env = pkgs.mkShell {
+          buildInputs = with pkgs; [ curl ];
+
+          # we use SHELL to keep the host's shell
+          shellHook = ''
+            $SHELL
+          '';
```

Comment:
This is a neat trick, but I wouldn't recommend it without knowing more about it. Has this been documented somewhere?

```suggestion
```

If you do use this somewhere, I would recommend at least `exec $SHELL` so you don't get returned into a bash and/or the bash shell doesn't stick around. Maybe you need to pass arguments in order to support some `nix develop` features?

I'm not a big fan of hacks like this because `nix develop` itself is already a huge hack. It (and `nix-shell`) try to cover too many use cases, which have incompatible requirements, leading to unsolvable problems and/or a massive amount of options and corner cases. Other shells are _probably_ better covered by something involving `nix print-dev-env`, but that still relies on shell derivations and bash, whereas derivations are not meant to describe shells. I've proposed alternatives, but the implementation work is yet to be done. (and is not something I can allocate for)

Reply:
Offtopic, but - to use the host's shell, I'd highly recommend going the direnv route instead, especially with starship (you get a nice visual indication): https://haskell.flake.page/direnv

Reply:
That's a great suggestion, and `nix print-dev-env`-based, or based on the same principle, certainly because of how direnv works. Indeed I don't think we should mention that here. It's more something for the site, where we could have a guide for creating a more advanced shell setup.

Reply:
I will remove it. I agree with your points. I do use starship and using `SHELL` helps me keep it. Otherwise it "feels" strange to me, because as a user I expect to have some packages available, not my shell to change (but I don't have much idea what's happening internally haha). I'm gonna try the `exec $SHELL` because having to exit twice the shell it's annoying haha I've been wondering how to do it.

Going back to the discussion, I'll remove the `$SHELL` and I'm gonna dig a bit into the direnv example. Thanks for the link!